compiler/annotation/match_inference.rs

```diff
@@ -254,16 +254,11 @@ impl TypeInferenceGraph<'_> {
         &self,
         variable_registry: &VariableRegistry,
     ) -> Result<(), TypeInferenceError> {
-        let mut thing_variable_present = false;
-        let mut any_variable_empty = false;
-        self.vertices.annotations.iter().filter_map(|(var, types)| var.as_variable().map(|v| (v, types))).for_each(
-            |(var, types)| {
-                thing_variable_present =
-                    thing_variable_present || variable_registry.get_variable_category(var).unwrap().is_category_thing();
-                any_variable_empty = any_variable_empty || types.is_empty();
-            },
-        );
-        if any_variable_empty && thing_variable_present {
+        let thing_variable_present = self.vertices.annotations.iter().filter_map(|(var, _)| var.as_variable())
+            .any(|var| variable_registry.get_variable_category(var).unwrap().is_category_thing());
+
+        let any_vertex_empty = self.vertices.annotations.iter().any(|(_, types)| types.is_empty());
+        if any_vertex_empty && thing_variable_present {
             return Err(TypeInferenceError::DetectedUnsatisfiablePattern {});
         }
```

Comment:
Will this still break if we get a schema query with no annotations for a label vertex?

Reply:
No, because we'll actually replace the whole constraint with "Unsatisfiable".
It does feel a little brittle when you put it that way, so maybe we should update executor (later) to be resilient to empty annotations.